src/azul/es.py

```diff
@@ -1,25 +1,29 @@
+import logging
+from functools import lru_cache
+
+from aws_requests_auth.boto_utils import BotoAWSRequestsAuth
 from elasticsearch import Elasticsearch, RequestsHttpConnection
-from aws_requests_auth import boto_utils
-from aws_requests_auth.aws_auth import AWSRequestsAuth
 
 from azul import config
 from azul.deployment import aws
 
 
-def es_client():
-    host = config.es_endpoint[0]
-    port = config.es_endpoint[1]
-    ssl_required = True if port == 443 else False
-    if ssl_required:
-        aws_auth = AWSRequestsAuth(aws_host=host,
-                                   aws_region=aws.region_name,
-                                   aws_service='es',
-                                   **boto_utils.get_credentials())
-        return Elasticsearch(hosts=[{'host': host, 'port': port}],
-                             http_auth=aws_auth,
-                             use_ssl=True,
-                             verify_certs=True,
-                             connection_class=RequestsHttpConnection)
-    else:
-        return Elasticsearch(hosts=[{'host': host, 'port': port}],
-                             connection_class=RequestsHttpConnection)
+class ESClientFactory:
+    logger = logging.getLogger(__name__)
+
+    @staticmethod
```

Comment:
I wouldn't ask this normally but since you have to change the logger, I might as well: Please make this a `@classmethod` and …